Software/lib/GNSE_HAL/GNSE_hal.c

```diff
@@ -0,0 +1,152 @@
+/** Copyright © 2021 The Things Industries B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file GNSE_hal.c
+ *
+ * @copyright Copyright (c) 2021 The Things Industries B.V.
+ *
+ */
+
+#include "GNSE_hal.h"
+#include "app.h"
+#include "GNSE_acc.h"
+#include "GNSE_bm.h"
+
+/**
+  * @brief Initialises the Sensors for GNSE: the accelerometer and humidity/temperature sensor
+  * @note I2C interface should be initialised before this function
+  * @param none
+  * @return none
+  */
+void GNSE_Sensors_Init(void)
+{
+    GNSE_BSP_LS_Init(LOAD_SWITCH_SENSORS);
+    GNSE_BSP_LS_On(LOAD_SWITCH_SENSORS);
+    /* Delay required to let the sensors start up */
+    HAL_Delay(100);
+
+    GNSE_ACC_Init();
+    SHTC3_probe();
+}
+
+/**
+  * @brief Deinitialises the Sensors for GNSE: the accelerometer and humidity/temperature sensor
+  * @note The I2C interface is not turned off by this function
+  * @param none
+  * @return none
+  */
+void GNSE_Sensors_DeInit(void)
+{
+    GNSE_ACC_DeInit();
+    GNSE_BSP_LS_Off(LOAD_SWITCH_SENSORS);
+    GNSE_BSP_LS_DeInit(LOAD_SWITCH_SENSORS);
+}
+
+/**
+  * @brief Initialises hardware of the Flash IC
+  * @param none
+  * @return none
+  */
+void GNSE_Flash_Init(void)
+{
+    GNSE_BSP_LS_Init(LOAD_SWITCH_FLASH);
+    GNSE_BSP_LS_On(LOAD_SWITCH_FLASH);
+    GNSE_BSP_Flash_SPI_Init();
+    MxSpi mxic = {0};
+    MxHardwareInit(&mxic);
+}
+
+/**
+  * @brief Deinitialises hardware of the Flash IC
+  * @param none
+  * @return none
+  */
+void GNSE_Flash_DeInit(void)
+{
+    GNSE_BSP_LS_Off(LOAD_SWITCH_FLASH);
+    GNSE_BSP_LS_DeInit(LOAD_SWITCH_FLASH);
+    GNSE_BSP_Flash_SPI_DeInit();
+}
+
+/**
+  * @brief Initialises all common GNSE hardware
+  * @param gnse_inits Interfaces to configure
+  * @return none
+  */
+void GNSE_Init(GNSE_Init_Cfg gnse_inits)
+{
+    if(gnse_inits.I2C_GNSE_INIT)
+    {
+        GNSE_BSP_Sensor_I2C1_Init();
+    }
+    if(gnse_inits.SENSORS_GNSE_INIT)
+    {
+        GNSE_Sensors_Init();
+    }
```

Comment:
At this level of abstraction the user doesn't care if it's I2C or SPI, and there is dedicated APIs for this.
There should be an init for internal sensors (I2C1) and an init for external sensors (I2C2)